src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -235,140 +203,35 @@ def try_resolve_urls(
-        ReleaseArtifactBundle.objects.filter(
-            organization_id=project.organization.id,
-            release_name=release_name,
-            # In case no dist is provided, we will fall back to "" which is the NULL equivalent for our tables.
-            # See `_create_artifact_bundle` in `src/sentry/tasks/assemble.py` for the reference.
-            dist_name=dist_name or "",
+def get_legacy_release_bundles(release: Release, dist: Optional[Distribution]):
+    return set(
+        ReleaseFile.objects.select_related("file")
+        .filter(
+            release_id=release.id,
+            dist_id=dist.id if dist else None,
+            artifact_count=0,
+            file__type=RELEASE_BUNDLE_TYPE,
         )
-        .select_related("artifact_bundle__file")
-        .order_by("-date_added")[:MAX_SCANNED_BUNDLES]
+        .values_list("file_id", flat=True)
+        .order_by("-file__timestamp")[:MAX_BUNDLES_QUERY]
     )
 
-    remaining_urls = urls
-
-    for release in releases_with_bundles:
-        file_id = release.artifact_bundle.file.id
-        file = release.artifact_bundle.file.getfile()
-
-        archive = ArtifactBundleArchive(file)
-        manifest = archive.manifest
-        archive.close()
 
-        def url_in_manifest(url: str):
-            if url_exists_in_manifest(manifest, url):
-                bundle_file_ids.add(file_id)
-                return True
-            return False
-
-        remaining_urls = [url for url in remaining_urls if not url_in_manifest(url)]
-
-        if not remaining_urls:
-            return remaining_urls
-
-    return remaining_urls
-
-
-def collect_legacy_artifact_bundles_containing_urls(
-    urls: List[str], release: Release, dist: Optional[Distribution], bundle_file_ids: Set[int]
-) -> List[str]:
-    artifact_index = None
-    try:
-        artifact_index = read_artifact_index(release, dist, artifact_count__gt=0)
-    except Exception as exc:
-        logger.error("Failed to read artifact index", exc_info=exc)
-    if not artifact_index:
-        return urls
-
-    artifact_archives = dict()
-
-    def url_in_any_artifact_index(url: str):
-        file = find_file_in_archive_index(artifact_index, url)
-        if file is not None:
-            ident = file["archive_ident"]
-            archive = artifact_archives.get(ident)
-
-            if archive is not None:
-                bundle_file_ids.add(archive)
-            else:
-                artifact_file_id = ReleaseFile.objects.filter(
-                    release_id=release.id, ident=ident
-                ).values("file_id")[0]["file_id"]
-                artifact_archives[ident] = artifact_file_id
-                bundle_file_ids.add(artifact_file_id)
-            return True
-        return False
-
-    return [url for url in urls if not url_in_any_artifact_index(url)]
-
-
-def get_releasefiles_matching_urls(urls: List[str], release: Release) -> Sequence[ReleaseFile]:
+def get_releasefiles_matching_url(
```

Comment:
```suggestion
def get_legacy_releasefile_by_file_url(
```
I would change the name to be a bit more indicative that we are getting a legacy `ReleaseFile` by the url of the file.